### PR TITLE
When reseting an host network: keep the primary address type

### PR DESCRIPTION
**Pull request by benjamreis**
See: https://github.com/xapi-project/xen-api/pull/6008

### CODE REVIEW

**Comment by benjamreis** [comment]
[v8.3-incoming scratch build](https://koji.xcp-ng.org/taskinfo?taskID=65976)

**Comment by stormi** [comment]
There's a comment upstream with another suggestion for the fix, so let's wait a bit.

**Comment by benjamreis** [comment]
Addressed and merged

**Comment by benjamreis** [comment]
[new scratch build towards v8.3-incoming](https://koji.xcp-ng.org/taskinfo?taskID=66273)

**Comment by stormi** [comment]
What about the PR name and commit message?

**Comment by stormi** [comment]
(and patch name)

**Comment by benjamreis** [comment]
Its still relevant since this is fixing when reseting the network.

File: SOURCES/xapi-24.19.2-keep-address-type-network-reset.XCP-ng.patch [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+index 56dff77924..3df1d692b3 100644
+--- a/ocaml/xapi/xapi_pif.ml
++++ b/ocaml/xapi/xapi_pif.ml
+@@ -475,6 +475,10 @@ let introduce_internal ?network ?(physical = true) ~t:_ ~__context ~host ~mAC
+   let capabilities = Net.Interface.get_capabilities dbg device in
+   let pci = get_device_pci ~__context ~host ~device in
+   let pif = Ref.make () in
++  let primary_address_type =
++    Record_util.primary_address_type_of_string
++      (Xapi_inventory.lookup Xapi_inventory._management_address_type)
```
[the remark this answers is not in the document]
Comment (gthvn1) [reply SOURCES/xapi-24.19.2-keep-address-type-network-reset.XCP-ng.patch:24]: otherwise LGTM

**Review by gthvn1 — APPROVED** [review]

**Review by stormi — APPROVED** [review]

**Comment by benjamreis** [comment]
[v8.3-incoming build](https://koji.xcp-ng.org/taskinfo?taskID=66276)

**Pull request merged**